### PR TITLE
Add support for default response

### LINKED ISSUE #6828: Api Response should have a way to represent a default response type
Swagger \ OpenAPI has a way to represent a `default` error response for all errors. See https://swagger.io/docs/specification/describing-responses#default. We don't have a good way to represent this. For the time being, we're resorting to using `StatusCode=0` to indicate this.

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #6828

### CODE REVIEW

**Comment by dougbu** [comment]
Shouldn't this PR target `release/2.1`? The `master` change would be automated.

File: src/Microsoft.AspNetCore.Mvc.Core/ProducesDefaultResponseTypeAttribute.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+using Microsoft.AspNetCore.Mvc.ApiExplorer;
+using Microsoft.AspNetCore.Mvc.Formatters;
+
+namespace Microsoft.AspNetCore.Mvc
+{
+    /// <summary>
+    /// A filter that specifies the <see cref="System.Type"/> for all HTTP status codes that are not covered by <see cref="ProducesResponseTypeAttribute"/>.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false, Inherited = true)]
+    public class ProducesDefaultResponseTypeAttribute : Attribute, IApiDefaultResponseMetadataProvider
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Core/ProducesDefaultResponseTypeAttribute.cs]: sealed

File: test/Microsoft.AspNetCore.Mvc.Core.Test/ApiConventionTypeAttributeTest.cs [hunk 2]
```diff
@@ -2,6 +2,7 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Linq;
```
Comment (dougbu) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/ApiConventionTypeAttributeTest.cs]: How do changes to this class relate to #6828?

Comment (pranavkm) [reply test/Microsoft.AspNetCore.Mvc.Core.Test/ApiConventionTypeAttributeTest.cs]: The error message changed. Refactoring it so we don't have to change it any time we add a new attribute that we allow on a convention.

Comment (dougbu) [reply test/Microsoft.AspNetCore.Mvc.Core.Test/ApiConventionTypeAttributeTest.cs]: Ah, IOW the real change was buried 😈 
:ok:

**Pull request merged**